controllers/idler/idler_controller.go

```diff
@@ -112,20 +112,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(ctx, idler, r.setStatusFailed, err,
 			"failed to ensure idling '%s'", idler.Name)
 	}
-	// Find the earlier pod to kill and requeue. Otherwise, use the RequeueTimeThresholdSeconds to requeue.
-	nextTime := nextPodToBeKilledAfter(logger, idler)
-	after := time.Duration(RequeueTimeThresholdSeconds) * time.Second
-	if nextTime == nil || *nextTime > after {
-		logger.Info("requeueing for next pod to check", "after_seconds", after.Seconds())
-		return reconcile.Result{
-			Requeue:      true,
-			RequeueAfter: after,
-		}, r.setStatusReady(ctx, idler)
-	}
-	logger.Info("requeueing for next pod to kill", "after_seconds", nextTime.Seconds())
+	// Requeue in idler.Spec.TimeoutSeconds or RequeueTimeThresholdSeconds, whichever is sooner.
+	nextPodToKillAfter := nextPodToBeKilledAfter(logger, idler)
+	maxRequeDuration := time.Duration(RequeueTimeThresholdSeconds) * time.Second
+	idlerTimeoutDuration := time.Duration(idler.Spec.TimeoutSeconds) * time.Second
+	after := findShortestDuration(nextPodToKillAfter, &maxRequeDuration, &idlerTimeoutDuration)
```

Comment:
I like the idea of the `findShortestDuration` function, but wouldn't make more sense to move 
```
nextPodToKillAfter := nextPodToBeKilledAfter(logger, idler)
maxRequeDuration := time.Duration(RequeueTimeThresholdSeconds) * time.Second
idlerTimeoutDuration := time.Duration(idler.Spec.TimeoutSeconds) * time.Second
```
into `findShortestDuration` function? the variable don't seem to be used anywhere else in this method...

Reply:
makes sense, moved it into the function and renamed it to `findShortestRequeueDuration` in [bfc3cda](https://github.com/codeready-toolchain/member-operator/pull/630/commits/bfc3cdab10e2b06e72dbf287053a7f7188fad488)
also removed the log.Info in `nextPodToKillAfter` - thus only logs the shortest requeue duration in `findShortestRequeueDuration`